ui/src/app/services/leaderboard.service.ts

```diff
@@ -58,16 +58,27 @@ export class LeaderboardService {
       this.updatingGlobal = true;
       this.leaderboardGlobal = [];
       this._api.getUsersByScore(5, (users) => {
-        this._api.getUserScores(users, (score) => { this.leaderboardGlobal.push(score); if (this.leaderboardGlobal.length === users.length) { this.updatingGlobal = false; }});
+        this._api.getUserScores(
+          users.filter((a) => {
+            if (a.username === this._user.getCurrentUsername()) {
+              a.username = 'YOU'
+            }
+            return a;
+          }),
+          (score) => {
+            this.leaderboardGlobal.push(score);
+            if (this.leaderboardGlobal.length === users.length) { this.updatingGlobal = false; }
+          });
```

Comment:
Interesting. Consider adding this as a single line filter to the callback (i.e.,

```
if (score.username === this._user.username) score.username = "YOU";

Reply:
Changed logic. Tested and it works fine.

Reply:
see getUserScores in api.service. Logic is moved to callback.